### ISSUE TITLE
Getting HTTP 400 error when trying to enable the Firebase extension through CLI

### ISSUE THREAD

**Issue by jon-flowers** [typesense/firestore-typesense-search#28]
## Description

I am using the Firebase CLI to try to enable the extension for a Firebase project.  Major note is that the last few times I did this it worked fine, so this seems to be a new issue.  I think it may be related to this change: https://github.com/typesense/firestore-typesense-search/commit/44c5b37ce84ea2573cc40419f0e414865bbc4544

The `TYPESENSE_API_KEY` parameter appears to be the issue. Previously I had this set as the admin API key, which used to work. Now, when I try to enable the extension, I get an error related to secrets manager, which seems to be saying I have formatted it wrong.  Specifically it says it should be formatted like `projects/*/secrets/*/versions/*` instead of just the API key, but I am not sure exactly how to format my API key like that (I am not sure which each of the asterisks in this case should be)

## Steps to reproduce

Set up parameters file
Run Firebase command to enable extension

`firebase ext:install typesense/firestore-typesense-search --project=projectId --params=params.env --force`

## Expected Behavior

The extension will enabled for my Firebase project

## Actual Behavior

I receive a HTTP 400 error:

Error: HTTP Error: 400, Secret ID `123....` is invalid, must be of form `projects/*/secrets/*/versions/*`

## Metadata

**Typesense Version**: v0.23.0.rc27

**OS**: MacOS

**Comment by jasonbosco** [comment]
@jon-flowers Could you try installing via the Firebase web console? It did ask me to provision a new secret and I just had to paste the Typesense API key and click on "Create" in the UI.

@dackers86 Any thoughts on how to do this via the CLI?

**Comment by jon-flowers** [comment]
Thanks @jasonbosco , I checked and can confirm that I can use the UI to activate the extension.  The flow has definitely changed slightly, with the API key now asking me to create a secret, but if I click that it seems to create fine.  

That being said, based on the number of environments we have, we definitely would prefer to do this with the CLI / programmatically wherever possible

**Comment by jon-flowers** [comment]
Just checking in to see if there were any updates here, I'm blocked from rolling out Typesense to more environments unless I can automate the extension installation. Thank you!

**Comment by jon-flowers** [comment]
@jasonbosco  generally my file looks like this:

FIRESTORE_COLLECTION_PATH=receiptTemplates
FIRESTORE_COLLECTION_FIELDS=
TYPESENSE_HOSTS=abc.a1.typesense.net
TYPESENSE_API_KEY=abc123
TYPESENSE_COLLECTION_NAME=eng_test_receiptTemplates
LOCATION=useast1


The TYPESENSE_API_KEY value (filled with a generic string above) is just the actual API key that I get directly from the Typesense cloud console.  This definitely worked before, but if I need to change the format I am fine with that.

Were you able to get the above results by using a params.env file with the same formatting?  Or is that the interactive flow where you entered the API key in when it prompted for it?

**Comment by i14h** [comment]
@jon-flowers are you using the non-interactive flow `--force` option?

**Comment by jon-flowers** [comment]
> @jon-flowers are you using the non-interactive flow `--force` option? 

Yes I am

**Comment by i14h** [comment]
@joehan fyi

**Comment by i14h** [comment]
we think it's a combination of specifying the secret in the `params.env` file and using the `--force` param. But the error message should've been more clear. Let us investigate a little bit.

**Comment by jasonbosco** [comment]
Thanks @i14h. 

@jon-flowers While this is being investigated, could you use v0.2.7 of the extension, which doesn't use the Secret Manager: 

```
firebase ext:install typesense/firestore-typesense-search@0.2.7 --project=projectId --params=params.env --force
```

**Comment by jon-flowers** [comment]
@i14h got it, thank you!

@jasonbosco that makes sense, I will do that and that should unblock me for now.  Appreciate the help!

**Comment by i14h** [comment]
I strongly suggest using the more secure version. Try not specifying your secret in the params.env file and see if that makes the problem go away? We don't suggest storing secrets in plain text files that could end up checked into your source repo

**Comment by jon-flowers** [comment]
@i14h Sorry I probably should have added more context here.

I am using Terraform (with the third-party Typesense provider, as well as Google and local-execs) to build out a Typesense collection, alias, enable the extension for an existing Firestore database, and create the backfill sync document. 

Everything is pretty straightforward EXCEPT installing this extension, because the Google/Firebase providers do not have a Terraform resource for that. So I am using a local-exec instead.  Though I am actually using two local-execs:

1. The first one builds a simple params.env file, using dynamic values.  Long-term, these are stored in either an env.yaml (for non-secret parameters) or an env-sensitive.yaml (for secret parameters).  The env.yaml is maintained in the Terragrunt source code, the sensitive yaml is maintained in S3 and retrieved every time Terragrunt runs.  The params.env file is built within Terragrunt and so is only maintained temporarily in the cache before being deleted.
2. The second local-exec using the built params.env file to run the Firebase CLI command that installs the extension for my given Firebase/Firestore app/database.

I need to add the Secret in there somehow because otherwise the CLI asks for it to be input, and I need this to be non-interactive.  I am using Terragrunt/Terraform to deploy this to 20+ environments (and maintain those going forward), so i am trying to avoid having to manually run CLI commands for each environment, especially as we continue to add more in the future.  I know that if I don't specify the secret it will ask me for it (and the flow works, with it correctly converting the API key to a secret), but this will not work in CircleCI/Terragrunt

**Comment by jon-flowers** [comment]
> You should be able to create the secret manually on the Google Cloud Console in secrets manager or through the API then reference in the .env file like so:
> 
> ```
> TYPESENSE_API_KEY=projects/${param:PROJECT_NUMBER}/secrets/<secretName>/versions/latest
> ```

I *think* I tried something similar to that, I agree it should work but if I remember correctly the problem was that the extension is not expecting that format; basically it seems like the extension at least is expecting the API key itself in one place and the GCP format you used above in another place, meaning it fails no matter what.  Assuming I am remembering this correctly, when I tested it using the above method I got a different error that was due to the formatting.

This *sort of* makes sense, since the UI / manual flow to enable the extension actually does ask for first the API Key itself, then prompts you to click a button to create the secret, and then uses that secret in the config.  So when you use the CLI method to enable the extension it's probably trying to do at least two things:

1. Using TYPESENSE_API_KEY to generate a new secret
2. Using that new secret in the config

And unfortunately it's not possible to format TYPESENSE_API_KEY in a way that works for both those steps.

**Comment by PMLS3** [comment]
Has anyone got a solution for this? First time using the extension and also getting  Request failed with HTTP code 404 | Server said: Not Found... I've set mine up through the Firebase Web Ui...

**Comment by jasonbosco** [comment]
@PMLS3 This issue is referencing an HTTP 400 error, when installing this extension via the Firebase CLI. 

Your HTTP 404 error is one that comes from Typesense and is unrelated to this issue. You're seeing that error most likely because you need to setup a collection in Typesense first - the extension does not create the Typesense collection for you. See Step 3 [here](https://github.com/typesense/firestore-typesense-search#step-1%EF%B8%8F%E2%83%A3--setup-prerequisites). If that's not the issue, and you do have a collection setup in Typesense, could you open a separate issue?

**Comment by tharropoulos** [comment]
Closing this due to inactivity for an extended period. Feel free to open a new issue if this is still relevant, ideally with updated context and requirements given how much time has passed.
